public/javascript/idea.js

```diff
@@ -1,59 +1,83 @@
 class Idea {
 
-  constructor(x, y, color, text) {
-  this.x = x || 10;
-  this.y = y || 10;
-  this.color = color || "#76A9DC";
+  constructor(x, y, color, shape, text) {
+    this.shape = shape;
+    this.x = x || 10;
+    this.y = y || 10;
+    this.color = color || "#76A9DC";
+    this.shape = shape;
 
-  // Rectangle
-  this.w = 200;
-  this.h = 95;
+    // Specifications for rectangles
+    this.w = 200;
+    this.h = 95;
 
-  // Cirlce
-  this.radius = 70;
-  this.startAngle = 0;
-  this.endAngle = 2 * Math.PI;
-
-  //text box
-  this.text = text;
-  this.font = 'bold 24px Courier';
-  this.fontColor = 'white';
+    // Specifications for circles
+    this.radius = 70;
+    this.startAngle = 0;
+    this.endAngle = 2 * Math.PI;
 
+    // Specifications for text
+    this.text = text;
+    this.font = 'bold 24px Courier';
+    this.fontColor = 'white';
   }
+
   contains(mx, my) {
-    return(this.x <= mx) && (this.x + this.w >= mx) && (this.y <= my) && (this.y + this.h >= my);
+    if (this.shape === 'rectangle') {
+      return(this.x <= mx) && (this.x + this.w >= mx) && (this.y <= my) && (this.y + this.h >= my);
+    } else {
+      let distancesquared = (this.x - mx) * (this.x - mx) + (this.y - my) * (this.y - my);
+      return distancesquared <= this.radius * this.radius;
+    }
   }
 
   draw(context) {
-    context.fillStyle = this.color;
-    context.fillRect(this.x, this.y, this.w, this.h);
-    context.textBaseline = 'middle';
-    context.font = this.font;
-    context.fillStyle = this.fontColor;
-    textX = this.x + this.w/2 - context.measureText(this.text).width/2;
-    textY = this.y + this.h/2;
-    context.fillText(this.text, textX, textY);
+    if (this.shape === 'rectangle') {
+      context.fillStyle = this.color;
+      context.fillRect(this.x, this.y, this.w, this.h);
+      context.textBaseline = 'middle';
+      context.textAlign = 'center';
+      context.font = this.font;
+      context.fillStyle = this.fontColor;
+      textX = this.x + (this.w / 2);
+      textY = this.y + (this.h / 2);
+      context.fillText(this.text, textX, textY);
```

Comment:
Not very well, no. Basically, with a little trial and error, I found that this works just as well as how we had it before. At least on my machine!